packages/marshal/src/types.js

```diff
@@ -169,17 +169,33 @@ export {};
  * @property {Slot[]} slots
  */
 
+/**
+ * @callback Validator
+ * @param {string} key
+ * @param {Passable} value
+ */
+
+/**
+ * @callback MakeValidator
+ * @param {boolean} jsonPass - true if the validator is to be used in the JSON
+ * replacer/reviver.  Otherwise, used when encoding/decoding the contents of the
```

Comment:
This surprises me. Without looking yet at the actual code which uses it, I'd expect either
   * these two cases are sufficiently similar that no flag is needed, or
   * these cases are different enough that they should be represented as distinct functions.

Of course, some cases fall in the middle ground where a flag is the right compromise. But I don't understand if this is one of those.

Reply:
I choose distinct functions, an encoding pass and a JSON pass.  Will implement that, if you agree.

Reply:
Actually, just `serialize(Passable, inputValidator)` and `unserialize(capdata, outputValidator)`.  The JSON passes can be done before or after serialization without losing  basic functionality.